drgn/helpers/linux/cgroup.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Cgroup
+------
+
+The ``drgn.helpers.linux.cgroup`` module provides helpers for working with
+cgroup interface in :linux:`include/linux/cgroup.h`. Only cgroup v2 is
+supported.
+"""
+
+import os
+
+from drgn import Object, container_of
+
+
+__all__ = [
+        'cgroup_name',
+        'cgroup_parent',
+        'cgroup_path',
+        'sock_cgroup',
+]
+
+
+def sock_cgroup(skcd):
```

Comment:
The kernel has `sock_cgroup_ptr`, how about we make this return a `struct cgroup *` instead of `struct cgroup` and rename it for consistency?

Reply:
Lol, that what I initially did, but then I decided to make it "more python-friendly". Sure, can convert it back to `_ptr` version.